app/models/repo.js

```diff
@@ -136,17 +140,27 @@ const Repo = VcsEntity.extend({
     return this._buildObservableArray(builds);
   }),
 
-  pullRequests: computed('id', function () {
-    let id = this.id;
-    const builds = this.store.filter('build', {
-      event_type: 'pull_request',
-      repository_id: id,
-    }, (b) => {
-      const isPullRequest = b.get('eventType') === 'pull_request';
-      return this._buildRepoMatches(b, id) && isPullRequest;
-    });
-    return this._buildObservableArray(builds);
-  }),
+  // Using dynamicQuery effectively relies on accurate pagination, which
+  // is not possible if skip_count=true, which is the default for queries in
+  // the build adapter. force_count allows overriding this behavior, but we
+  // should use it cautiously for performance reasons. I think it may
+  // be fine for pull requests though?
+  fetchBuilds({ page, filter, eventType, forceCount }) {
+    const { id: repoId, store } = this;
+    const offset = (page - 1) * limit;
+
+    return store.paginated('build', {
```

Comment:
Replacing `store.filter` by `store.paginated` may cause issues when builds are pushed to the store by Pusher. I'm not entirely sure, but `filter` seems to be something dynamic and updates its contents when store contents change, whereas `paginated` just returns whatever exists in the store at the moment and never updates unless called explicitly. But I may be wrong here, it's worth checking as a _possible_ issue 😄 Even if I'm right, maybe Pull Requests list doesn't actually need to be dynamic? 🤔

Reply:
That's a good point, I'll look into it a bit

Reply:
I think you're right, this results in store changes from pusher to not be loaded. I tried switching back to `store.filter` but leaving in the other stuff, but it didn't help. And I tried a few other things as well, but everything I've tried so far either results in the original issue, won't load pusher updates, or only loads a single build until you click to show more.

I'd prefer not to lose the live updates from pusher, so we may need a different approach. I might try a few other things tomorrow, but if you have any suggestions, let me know.

Although I believe that ember 3.13+ breaks some of this same stuff, so if that's coming up soon anyway, we could hold on this until then. This issue of the missing notice showing on initial load is not specific to the vcs update, and seems to have existed for a while.